### DISCUSSION TITLE
mcdev createDeltaPkg issue - ❌  No changes found. Check what branch you are currently on and if the target branch

### DISCUSSION THREAD

**Discussion by narmadakankuri9** [Accenture/sfmc-devtools#1378] [Q&A] [answered]
Hello!

I have retrieved the all DataExtension from DevBU and I committed to related dev branch as well. 
Now I want to create createDeltaPkg for those retrieved extensions and deploy to different org. But when I tried to create createDeltaPkg  it is giving an issue 

  mcdev cdp dev
15:21:29 info: Create Delta Package ::
15:21:29 info: Analyzing changes in directories: retrieve/xxxxxxxxxxx/xxxxxxx based on commit range: odev..HEAD
15:21:29 warn: - ❌  No changes found. Check what branch you are currently on and if the target branch name (dev converted to dev..HEAD) was correct

This is the process I followed:

- mcdev init - with dev org cred 
- retrieved the data extension
- committed to dev branch 
- mcdev init - for target branch
- committed to related target branch
- in the dev branch I'm trying to run   mcdev cdp dev, but it is giving the issue.

**Comment by JoernBerkefeld** [comment]
from your log entry i can see you likely ran `mcdev dev` without further details, correct?
have you switched over to your master branch before running this? otherwise it's basically comparing one and the same commit with each other, which cannot find any differences.

if the above didn't help:
could you please re-run the command and then share the latest .log file from your /logs/ folder?
it might contain credential/bu names -- in case you want to obfuscate those before uploading the file

**Comment by narmadakankuri9** [comment]
Here is the process:

- ran mcdev init on dev org --> given all credential details of dev org --> retrieved the data extension from dev org --> committed into dev branch
-  ran mcdev init on target org -->given all credential details of target org --> committed into target branch
- Now in my local I have both dev branch and target branch. 
- Again Made some changes in dev org --> retrieved that into local --> committed into dev branch 
- Now checkout to hotfix branch and ran the mcdev cdp dev..hotfix
- Even tried running mcdev cdp hotfix..dev

Still getting the same issue as No changes found.

**Comment by JoernBerkefeld** [comment]
@JoernBerkefeld note to self

**Comment by narmadakankuri9** [comment]
Hello @JoernBerkefeld  - Now looks like it is trying to fetch the difference and creating the package but I do see some issues.

 mcdev createDeltaPkg XXXXX../XXXX
17:58:15 info: Create Delta Package ::
17:58:15 error: Deployment Target: Market List XXXXXXX/XXXX is not defined
17:58:15 info: Analyzing changes in directories: retrieve/XXXXX/XXXXX based on commit range: XXXXX..XXXX
17:58:15 info: - ✔️  Identified changes: Add/Update=1, Move=0, Delete=0
17:58:15 info: Saved report in docs\deltaPackage\delta_package.md
17:58:15 info: =============
17:58:15 info: Retrieve template from Git delta
17:58:15 info: ⚡ mcdev bt XXXXXXX/XXXXX dataExtension "XXXX" XXXX
17:58:15 error: **Could not find the market 'G' in your configuration file.**
17:58:15 info: Available markets are: XXXXXX
17:58:15 info: - ✔️  Templates created: 1
17:58:15 info: =============
17:58:15 info: Build deploy definitions from delta templates
? Do you want to empty the deploy folder (ensures no files from previous deployments remain)? yes
17:59:06 info: ⚡ mcdev bdb XXXX/XXXX dataExtension "XXXX"
17:59:06 error: Market List XXXX/XXXXXX is not defined
17:59:06 info: - ✔️  Deploy definitions created
17:59:06 info: You can now run deploy on the prepared BUs

**Comment by JoernBerkefeld** [comment, marked as answer]
hi @narmadakankuri9  - if you want to use cdp without templating (without markets), then you CAN use the --filter option instead, which takes the folder name BELOW the retrieve/ folder.

The error createDeltaPackage relies on the configuration of `options.deployment.sourceTargetMapping` in your `.mcdevrc` config-file

By default that is set to 

```json
{
        "deployment-source": "deployment-target"
}
```

You then have to make sure that you have the markets defined in your config, similar to whats in the big example in the wiki: https://github.com/Accenture/sfmc-devtools/wiki/06.d-~-Git-based-deployments-and-CICD

```json
  "markets": {
    "dev": {},
    "qa": {},
    "prod": {}
  },
  "marketList": {
    "deployment-source": {
      "description": "Define one 1:1 BU-Market combo here to as source for automated creation of deployment packages; you can create more than one source market list",
      "MyProject/DEV": "dev"
    },
    "deployment-target": {
      "description": "Define n BU-Market combo here to as target for automated creation of deployment packages; you can create more than one target market list and they can be as complex as you like",
      "MyProject/QA": "qa",
      "MyProject/PROD": "prod"
    }
  }
}
```

For more details on how to define markets and marketLists:

https://github.com/Accenture/sfmc-devtools/wiki/07.-Advanced-Configuration#market-configuration
